plugins/generator-1.19.2/forge-1.19.2/templates/item/item.java.ftl

```diff
@@ -252,5 +255,97 @@ public class ${name}Item extends Item {
 			stack.getCapability(ForgeCapabilities.ITEM_HANDLER, null).ifPresent(capability -> ((ItemStackHandler) capability).deserializeNBT((CompoundTag) nbt.get("Inventory")));
 	}
 	</#if>
+
+	<#if hasProcedure(data.onStoppedUsing) || (data.enableRanged && !data.shootConstantly)>
+		@Override public void releaseUsing(ItemStack itemstack, Level world, LivingEntity entity, int time) {
+			double x = entity.getX();
+			double y = entity.getY();
+			double z = entity.getZ();
+			<#if hasProcedure(data.onStoppedUsing)>
+				<@procedureCode data.onStoppedUsing, {
+					"x": "x",
+					"y": "y",
+					"z": "z",
+					"world": "world",
+					"entity": "entity",
+					"itemstack": "itemstack",
+					"time": "time"
+				}/>
+			</#if>
+			<#if data.enableRanged && !data.shootConstantly>
+				if (!world.isClientSide() && entity instanceof ServerPlayer player) {
+					if (<@procedureOBJToConditionCode data.useCondition/>) {
+						<@arrowShootCode/>
+					}
+				}
+			</#if>
+		}
+	</#if>
+
+	<#if data.enableRanged && data.shootConstantly>
+		@Override public void onUsingTick(ItemStack itemstack, LivingEntity entity, int count) {

```

Comment:
This method is deprecated

Reply:
Can you tell me where you saw the method is deprecated? The code does not say it's deprecated

![image](https://github.com/MCreator/MCreator/assets/38427877/02e2d744-5af4-4b65-8c29-1667c18f5996)

Reply:
This looks deprecated to me:

![image](https://github.com/MCreator/MCreator/assets/16374228/596a8617-26f8-4436-a622-12bc88d31c94)

Please when working on new features, run tests locally and inspect the logs they provide as testing is not as simple as a "green checkmark" on GH PR, all good

Reply:
If this method is deprecated, I wonder what is the good method because Forge wrote notbing about it in its code (and this is why I used this method)...

Reply:
Ok, I found the problem. It was Forge, but you with your review. The method is only marked as deprecated since 1.19.4, but you wrote your comment in the 1.19.2 generator, so I thought both versions had this method deprecated (and I checked the code in 1.19.2), while it's only since Forge 1.19.4.

Reply:
Sorry, my bad